src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -645,12 +646,13 @@ double DecisionTree<FitnessFunction,
         dimGain = CategoricalSplit::template SplitIfBetter<UseWeights>(bestGain,
             data.cols(begin, begin + count - 1).row(i),
             datasetInfo.NumMappings(i),
-            labels.subvec(begin, begin + count - 1),
+            labels,
+            begin,
```

Comment:
Is there any particular reason to not use a `subvec()`?  (I suppose it may be better to create an alias with the advanced `vec` constructor, so that there is no data copying.)

Reply:
I was getting this weird template error when I was passing a `subvec()`.
`candidate template ignored: could not match 'Row' against 'subview_row'`. 
I had a long discussion with jonpsy about this on April 9 on IRC. But we couldn't figure it out. 


> (I suppose it may be better to create an alias with the advanced `vec` constructor, so that there is no data copying.)

Correct me if I am wrong here. I am passing `labels` as `const arma::Row<ElemType>& labels`. Since I am passing it by reference, it shouldn't be copying it right?

Reply:
Yes, but actually that is the cause of the error. :)  If you call `.subvec()`, I believe that returns a subview, not an `arma::Row<ElemType>`.  So, if you templatize to accept any `LabelsType` like was discussed in a different comment, that could fix the issue.  Alternately, you can use the 'advanced' constructor to construct an alias of the `labels` vector of the correct length, starting at the correct position.

Reply:
Okay, I will give it another shot and see if I am successful with templates. If not, then I will do it with an advanced constructor.